Replace idle call to finalize_new_views() with immediate call

c4eater opened the pull request:
## Summary
This change adds a new (failing) test demonstrating #1232 and then a fix, which includes making a synchronous call to finalize_new_views.

## Related Issues
Closes #1232.
Closes #1115.

## Checklist

- [x] I have responded to reviews and made changes where appropriate.
- [x] I have tested the code with `cargo test --all` / `./rust/run_all_checks`.
- [x] I have updated comments / documentation related to the changes I made.
- [x] I have rebased my PR branch onto xi-editor/master.

Cogitri commented:
Seems like this currently breaks opening files during startup, e.g. if you open tau with a file as argument xi-core crashes now.

c4eater commented:
> Seems like this currently breaks opening files during startup, e.g. if you open tau with a file as argument xi-core crashes now.

OK you were probably trying my earlier (invalid) implementation (c4eater/xi-editor@1e60c5a)...
I just rebased against the xi-editor's master and got a green CI status. Have no pb opening a file (with a different frontend) as well. Can you please check it again?

Cogitri commented:
Just tried again, seems to still be happening :/

Cogitri commented:
Ah, so it seems like without this the cmd flow is like this:

1) tau -> xi: client_started & config_changed
2) xi -> tau: available_themes & available_languages
3) tau -> xi: set_theme and language

c4eater commented:
Aha, I get it. My fix moved "EventContext::update_wrap_settings" (which in turn could ask for "measure_width") from the idle handler into an immediate call to do_new_view. This is probably legal but I don't need to change the behavior that dramatically to fix the bug. I've updated my PR once again and I believe that the behavior change now is truly minimalistic - only view::update_wrap_settings is now called immediately.

Cogitri commented:
This also fixes #1115 too apparently, could you add that to the commit msg please? :)

LGTM otherwise

c4eater commented:
OK, updated the commit msg.

Cogitri commented:
I think `(fixes #1232, #1115)` doesn't work, you need to do `fixes #1232, fixes #1115`

c4eater commented:
Interesting... Just played around with the commit msg and found that the issue page and the presence of "maybe fixed by..." probably reacts on what is written in the PR description, not in the commit (s). I've updated the commit msg to "fixes #x, fixes #y" but that didn't affect 1115's description page. It picked up the changes I updated the PR text though.

Anyway, looks like the comment is consistent now.

Cogitri commented:
Thank you!
